webapp/src/test/java/com/box/l10n/mojito/service/tm/TextUnitBatchMatcherTest.java

```diff
@@ -516,9 +579,89 @@ public void testCreateMatchByNameCommentAndUnused_DoesNotMatchByComment() {
+            createUnusedPluralTextUnitDTO("name-4", "one", "comment-4"),
+            createUnusedPluralTextUnitDTO("name-4", "zero", "comment-4"),
+            createUnusedPluralTextUnitDTO("name-4", "two", "comment-4"),
+            createUnusedPluralTextUnitDTO("name-4", "few", "comment-4"),
+            createUnusedPluralTextUnitDTO("name-4", "many", "comment-4"));
+
     Function<TextUnitForBatchMatcher, List<TextUnitDTO>> textUnitForBatchMatcherListFunction =
-        textUnitBatchMatcher.matchByNameAndPluralPrefix(Collections.emptyList(), PLURAL_SEPARATOR);
-    textUnitForBatchMatcherListFunction.apply(createTextUnitForBatchMatcher("test", "source text"));
+        textUnitBatchMatcher.matchByNameAndPluralPrefix(existingTextUnitDTOs, PLURAL_SEPARATOR);
+
+    List<TextUnitDTO> noMatchByNameSourceAndUsed =
+        textUnitForBatchMatcherListFunction.apply(
+            createTextUnitForBatchMatcher("name-1", "Second string"));
+    assertTrue(noMatchByNameSourceAndUsed.isEmpty());
+
+    List<TextUnitDTO> matchByNameSourceAndUsed =
+        textUnitForBatchMatcherListFunction.apply(
+            createTextUnitForBatchMatcher("name-1", "First string"));
+    assertFalse(matchByNameSourceAndUsed.isEmpty());
+    assertEquals("name-1", matchByNameSourceAndUsed.getFirst().getName());
+
+    List<TextUnitDTO> secondMatchByNameSourceAndUsed =
+        textUnitForBatchMatcherListFunction.apply(
+            createTextUnitForBatchMatcher("name-1", "First string"));
+    assertTrue(secondMatchByNameSourceAndUsed.isEmpty());
+
+    List<TextUnitDTO> noMatchByNameSourceAndUnused =
+        textUnitForBatchMatcherListFunction.apply(
+            createTextUnitForBatchMatcher("name-2", "First string"));
+    assertTrue(noMatchByNameSourceAndUnused.isEmpty());
+
+    List<TextUnitDTO> matchByNameSourceAndUnused =
+        textUnitForBatchMatcherListFunction.apply(
+            createTextUnitForBatchMatcher("name-2", "Second string"));
+    assertFalse(matchByNameSourceAndUnused.isEmpty());
+    assertEquals("name-2", matchByNameSourceAndUnused.getFirst().getName());
+
+    List<TextUnitDTO> secondMatchByNameSourceAndUnused =
+        textUnitForBatchMatcherListFunction.apply(
+            createTextUnitForBatchMatcher("name-2", "Second string"));
+    assertTrue(secondMatchByNameSourceAndUnused.isEmpty());
+
+    List<TextUnitDTO> noMatchByNameCommentAndUsed =
+        textUnitForBatchMatcherListFunction.apply(
+            createPluralTextUnitForBatchMatcher("name-3", "comment-4"));
+    assertTrue(noMatchByNameCommentAndUsed.isEmpty());
+
+    List<TextUnitDTO> matchByNameCommentAndUsed =
+        textUnitForBatchMatcherListFunction.apply(
+            createPluralTextUnitForBatchMatcher("name-3", "comment-3"));
+    assertFalse(matchByNameCommentAndUsed.isEmpty());
+    assertEquals("name-3_other", matchByNameCommentAndUsed.getFirst().getName());
+
+    List<TextUnitDTO> secondMatchByNameCommentAndUsed =
```

Comment:
Just to confirm this to test that if you matched once, you won't match the same text unit again?